vendor/k8s.io/kubernetes/staging/src/k8s.io/kube-gen/cmd/client-gen/generators/util/tags.go

```diff
@@ -146,16 +189,83 @@ func ParseClientGenTags(lines []string) (Tags, error) {
 		}
 		ret.SkipVerbs = skipVerbs
 	}
+	var err error
+	if ret.Expansions, err = parseClientExpansions(values); err != nil {
+		return ret, err
+	}
 	return ret, validateClientGenTags(values)
 }
 
+func parseClientExpansions(tags map[string][]string) ([]expansion, error) {
+	ret := []expansion{}
+	for name, values := range tags {
+		if !strings.HasPrefix(name, "genclient:") || !strings.HasSuffix(name, "Verb") {
+			continue
+		}
+		name = strings.TrimPrefix(name, "genclient:")
+		exp := expansion{}
+		for _, supportedVerb := range SupportedVerbs {
+			if strings.HasPrefix(name, supportedVerb) {
+				exp.VerbType = supportedVerb
+				break
+			}
+		}
+		if len(exp.VerbType) == 0 {
+			return nil, fmt.Errorf("invalid expansion verb type prefix %q (must be one of %s)", name, strings.Join(SupportedVerbs, ","))
+			continue
+		}
+		exp.VerbName = strings.TrimSuffix(strings.TrimPrefix(name, exp.VerbType), "Verb")
+		if len(values[0]) > 0 {
```

Comment:
are we sure there are values?

Reply:
do I remember right that multiple tag line give you multiple values here?

Reply:
yes, if you repeat the same tag (same verb) with different configuration it will give you multiple values... i'm not sure if we want to support it... something like this looks like something we don't want to recommend?

```
// +genclient:createFooVerb=subresource=foo
// +genclient:createFooVerb=output=Foo
```

Reply:
nvmd. this actually works well if I move this into a for _, v := range values {} loop... so people can use this in multi-line if they really want